src/main/java/org/pac4j/vertx/auth/Pac4jUser.java

```diff
@@ -82,8 +82,8 @@ public int readFromBuffer(int pos, Buffer buffer) {
         final byte[] jsonBytes = buffer.getBytes(pos, pos + jsonByteCount);
         pos += jsonByteCount;
         final String json = new String(jsonBytes, StandardCharsets.UTF_8);
-        final UserProfile userProfile = (UserProfile) DefaultJsonConverter.getInstance().decodeObject(json);
-        setUserProfile(userProfile);
+        final UserProfile userProfileLocal = (UserProfile) DefaultJsonConverter.getInstance().decodeObject(json);
```

Comment:
Suggest something like decodedUserProfile to indicate that it's the UserProfile we've built by decoding from the buffer. That fairly clearly puts into context as well as segregating it from the member userProfile, wouldn't you agree?